### PR TITLE
W-17225272: Fix metadata key id warning

### PR DESCRIPTION
**Pull request by d4nielyan9**
Test result with mule app (basic-1.0.0-SNAPSHOT-mule-application.jar)
```
WARN  2024-12-04 09:09:09,027 [ForkJoinPool.commonPool-worker-2] [processor: ; event: ] org.mule.runtime.module.extension.internal.loader.utils.JavaMetadataKeyIdModelParserUtils: A Keys Resolver is being defined without defining an Output Resolver, Input Resolver nor Attributes Resolver for element deleteColumnsValue of extension CassandraDB
WARN  2024-12-04 09:09:22,321 [ForkJoinPool.commonPool-worker-2] [processor: ; event: ] org.mule.runtime.module.extension.internal.loader.utils.JavaMetadataKeyIdModelParserUtils: A Keys Resolver is being defined without defining an Output Resolver, Input Resolver nor Attributes Resolver for element deleteRows of extension CassandraDB
WARN  2024-12-04 09:09:32,250 [ForkJoinPool.commonPool-worker-2] [processor: ; event: ] org.mule.runtime.module.extension.internal.loader.utils.JavaMetadataKeyIdModelParserUtils: A Keys Resolver is being defined without defining an Output Resolver, Input Resolver nor Attributes Resolver for element select of extension CassandraDB
```

### CODE REVIEW

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/utils/JavaMetadataKeyIdModelParserUtils.java [hunk 1]
```diff
@@ -221,6 +221,13 @@ private static Optional<MetadataKeyModelParser> getKeyIdResolverModelParser(Outp
       }
     }
 
+    if (keyIdResolverModelParser.isPresent()
+        && outputResolverModelParser == null
+        && attributesResolverModelParser == null
+        && inputResolverModelParsers.isEmpty()) {
+      LOGGER.warn("A Keys Resolver is being defined without defining an Output Resolver, Input Resolver nor Attributes Resolver");
```
Comment (rbourbonnavarro) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/utils/JavaMetadataKeyIdModelParserUtils.java:228]: Remove the warning log from `getCategoryName` and add the `TODO W-14195099...` here.

Comment (rbourbonnavarro) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/utils/JavaMetadataKeyIdModelParserUtils.java:228]: The improvements for the log message are missing, see the partial work PR in the issue.

**Review by rbourbonnavarro — CHANGES_REQUESTED** [review]

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/utils/JavaMetadataKeyIdModelParserUtils.java [hunk 2]
```diff
@@ -225,7 +225,9 @@ private static Optional<MetadataKeyModelParser> getKeyIdResolverModelParser(Outp
         && outputResolverModelParser == null
         && attributesResolverModelParser == null
         && inputResolverModelParsers.isEmpty()) {
-      LOGGER.warn("A Keys Resolver is being defined without defining an Output Resolver, Input Resolver nor Attributes Resolver");
+      LOGGER
```
Comment (rbourbonnavarro) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/utils/JavaMetadataKeyIdModelParserUtils.java:228]: Remember to add the `TODO` that was associated to this log.

**Review by rbourbonnavarro — CHANGES_REQUESTED** [review]

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/utils/JavaMetadataKeyIdModelParserUtils.java (hunk 2 above)
Comment (d4nielyan9) [reply modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/utils/JavaMetadataKeyIdModelParserUtils.java:229]: sorry forgot to add this back!

**Comment by d4nielyan9** [comment]
--validate

**Review by rbourbonnavarro — APPROVED** [review]

**Comment by rbourbonnavarro** [comment]
--validate

**Comment by d4nielyan9** [comment]
--validate

**Pull request merged**